test/suite/test_layered32.py

```diff
@@ -83,41 +83,71 @@ def conn_extensions(self, extlist):
+        for k, v in kv.items():
             self.session.begin_transaction()
-            cursor[str(i)] = value1
+            cursor[k] = v
             if self.ts:
-                self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(5))
+                self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(ts))
             else:
                 self.session.commit_transaction()
+        cursor.close()
+
+    def test_layered_read_write(self):
+        self.session.create(self.uri, self.session_create_config())
 
+        # Populate the table with nitems.
+        inital_value = "abc" * 10
+        inital_ts = 5
+        kv = {str(i): inital_value for i in range(1, self.nitems + 1)}
+        self.insert(kv, inital_ts)
         self.session.checkpoint()
 
         # Re-open the connection to clear contents out of memory.
         new_config = self.conn_base_config + f'disaggregated=(checkpoint_meta="{self.disagg_get_complete_checkpoint_meta()}"),'
         self.reopen_conn(config=new_config)
 
-        cursor = self.session.open_cursor(self.uri, None, None)
-
-        # Perform a single update.
-        self.session.begin_transaction()
-        cursor[str(10)] = str(10 + 5 * 10) + str("abcdefghijklmnopqrstuvwxyz")
-        if self.ts:
-            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(10 + 5 * 10))
-        else:
-            self.session.commit_transaction()
-        self.session.checkpoint()
+        kv_modfied = {}
+        num_deltas = random.randint(1, 10)
+        for i in range(1, num_deltas + 1):
+            # Generate a random number of keys to insert.
+            random_key_range = random.randint(10, 2000)
+            kv = {
+                str(random.randint(1, self.nitems)): f"{i + j * i}abc"
+                for j in range(random_key_range)
+            }
+            # Insert random keys into the table.
+            self.insert(kv, inital_ts + i)
+            # Perform a checkpoint to write out a delta.
+            self.session.checkpoint()
+            # Merge kv into our cumulative dictionary
+            kv_modfied.update(kv)
 
         # Assert that we have written at least one internal page delta.
-        stat_cursor = self.session.open_cursor('statistics:', None, None)
-        rec_page_delta_leaf = stat_cursor[stat.conn.rec_page_delta_leaf][2]
-        self.assertGreater(rec_page_delta_leaf, 0)
-        rec_page_delta_internal = stat_cursor[stat.conn.rec_page_delta_internal][2]
-        self.assertGreater(rec_page_delta_internal, 0)
-        stat_cursor.close()
+        self.assertGreater(self.get_stat(stat.conn.rec_page_delta_leaf), 0)
+        self.assertGreater(self.get_stat(stat.conn.rec_page_delta_internal), 0)
+
+        # Re-open the connection to clear contents out of memory.
+        new_config = self.conn_base_config + f'disaggregated=(checkpoint_meta="{self.disagg_get_complete_checkpoint_meta()}"),'
```

Comment:
@Siddhartha8899 Can we keep the original simple test and create a separate test that is more complex?

Reply:
Updated in [4d73266](https://github.com/wiredtiger/wiredtiger/pull/11702/commits/4d73266d82442fded3915457b6e4200ce6022637)